scroll-search.lic

```diff
@@ -0,0 +1,520 @@
+    if $settings.nil? || $settings.empty?
+      echo("Settings was nil!") if $debug_mode
+    end
+
+    arg_definitions = 
+    [
+      [
+		{ name: 'search', regex: /^sea.*/i, optional: true, description: 'Searches given container and labels any unlabeled scrolls. ' },
+		{ name: 'list', regex: /^lis.*/i, optional: true, description: 'Lists all of the scrolls that have been categorized after a SEARCH. ' },
+        { name: 'container', regex: /\w+/i, optional: true, description: 'The container to search. ' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+	unless args.search || args.list
+		display_args(arg_definitions)
+		return
+	end
+
+    pause
+
+    if args.search
+      if args.container.nil?
+        echo "*** SPECIFY THE CONTAINER TO SEARCH! ***"
+        return
+      else
+        search_container(args.container)
+      end
+	elsif args.list
+      if @owned_scrolls.length <= 0
+        echo("*** NO SCROLLS HAVE BEEN ADDED! SEARCH YOUR CONTAINERS FIRST!***")
+        return
+      end
+
+      echo("Scroll scout is: #{@owned_scrolls.length}") if $debug_mode
+      display_scrolls()
+    else
+      echo("***INVALID ACTION***")
+    end
+  end
+
+  def search_container(container)
+    # return if invalid container
+
+	unless exists?(container)
+      echo("*** INVALID CONTAINER. ***")
+      return
+    end
+
+    # Get long name of container
+    $container_full = full_name(container)
+    $container = get_noun($container_full)
+
+    # iterate over all scroll types
+    scroll_types = $settings.scroll_nouns
+
+    if scroll_types.length <= 0 || scroll_types.nil?
+      echo("*** NO SCROLL TYPTES SET! SETUP YOUR YAML TO INCLUDE VALID \"scroll_types\". ***")
+      return
+    end
+
+    echo("Scroll type count is: #{scroll_types.length}.") if $debug_mode
+
+	# Clear current scroll listing from owned scrolls if
+	# user is searching an existing container
+	if @owned_scrolls.key? ($container_full)
+		echo("Clearing scroll array for this container, since it was previously searched --- #{$container_full}!") if $debug_mode
+		@owned_scrolls[$container_full] = []
+	end
+	
+	# reset scroll count
+	@scroll_count = 0
+	
+    scroll_types.each do |scroll|
+      echo "Searching for scrolls of type #{scroll}." if $debug_mode
+
+      scroll_list = get_scrolls(scroll)
+      next if scroll_list.empty?
+
+      look_scrolls(scroll_list)
+    end
+
+    # Save scrolls to user variables.
+	if UserVars.owned_scrolls.nil? && @owned_scrolls.nil?
+		echo("Creating an empty hash for vars!") if $debug_mode
+		UserVars.owned_scrolls = {}
+		return
+	end
+	
+	existing_hash = if UserVars.owned_scrolls.nil? then @owned_scrolls.clone else UserVars.owned_scrolls.clone end
+	
+	new_hash = @owned_scrolls
+	existing_hash.each do |container, scroll_array|
+		if new_hash.key? (container)
+			echo("Existing hash already has this container: #{container}") if $debug_mode
+			next
+		else
+			new_hash[container] = scroll_array
+		end
+	end

```

Comment:
Although it seems like `existing_hash.merge!(new_hash)` might just do what you want to do.